modules/table/src/main/java/org/apache/ignite/internal/table/distributed/gc/GcUpdateHandler.java

```diff
@@ -147,7 +147,8 @@ private VacuumResult internalVacuum(HybridTimestamp lowWatermark, Locker locker,
             }
 
             try (Cursor<ReadResult> cursor = storage.scanVersions(rowId)) {
-                indexUpdateHandler.tryRemoveFromIndexes(binaryRow, rowId, cursor);
+                // TODO: IGNITE-21005 We need to choose only those indexes that are not available for transactions
+                indexUpdateHandler.tryRemoveFromIndexes(binaryRow, rowId, cursor, null);
```

Comment:
I tried to analyze the code and I think that there should be no problems here since we first try to take the lock by RowId and then only clean out the garbage.